### PR TITLE
stk::mesh::MeshBuilder Conversion

### PR DESCRIPTION
**Pull request by ddement**
stk::mesh::BulkData's constructor will be deprecated in favor of the stk::mesh::MeshBuilder factory object in the near future.  This change updates Nalu-Wind to use this new factory for all BulkData construction.

### CODE REVIEW

**Review by jrood-nrel — APPROVED** [review]

**Review by psakievich — CHANGES_REQUESTED** [review]
As a soft request I'd like to see moving from a raw pointer to a smart pointer in realm.  If that blows up into something big then we can back off it from this PR, but it looks pretty straightforward from what I see here.

As more of a question, @ddement is the long term goal for `stk::mesh::MeshBuilder` to unify `MetaData` and `BulkData`?  If so is there more we can do to accept that API in our infrastructure? I see hints of it throughout the unit test code.

**Comment by alanw0** [comment]
@psakievich I don't think we can completely unify BulkData and MetaData. In Sierra we have a couple of scenarios where a MetaData persists and the associated BulkData gets destroyed and a new one gets constructed and attached to the persisting MetaData.
However, it is not necessary to separately hold a BulkData and a MetaData. If you have a BulkData, you can always get a MetaData from it.

**Comment by psakievich** [comment]
@alanw0 it seems like it would make sense to stop carrying around meta and bulk data at the realm level in nalu-wind then, and just opt for a single `stk::mesh::MexhBuilder` or `BulkData` instead

**Comment by ddement** [comment]
@psakievich I just pushed up a version that stores BulkData as a shared_ptr.  Because of the way it's set in the unit test HelperObjects, using a unique_ptr would be a bit more work.  We could do it using move semantics from where the BulkDatas are created in the unit tests, but it might require rethinking the logic flow for some of them.  shared_ptr should work for now.  

Also, I removed Realm's separate storage of a MetaData pointer.  That pointer is only ever accessed/set inside Realm itself or inside the HelperObjects, and it is only ever set right next to where BulkData is being set.  BulkData now manages the lifetime of MetaData.  Instead, the meta_data() method on Realm just calls through to BulkData::mesh_meta_data().

**Review by psakievich — APPROVED** [review]
@ddement this looks great. Thanks for making that change.

**Comment by psakievich** [comment]
@tasmith4 let us know if the stk team doesn't have merge capabilities and who should have them from the stk team.

**Comment by alanw0** [comment]
@psakievich @jrood-nrel I think it would be good if all of the stk folks are added, keeping in mind that we will still adhere to practices like waiting for approvals, and passing CI if it's working, etc.

**Comment by psakievich** [comment]
@alanw0 not sure if you noticed on this PR but we have a bare bones CI running again thanks to the docker images from @eugeneswalker :)

**Pull request merged**
